packages/terminal/src/browser/terminal-preferences.ts

```diff
@@ -107,6 +107,33 @@ export const TerminalConfigSchema: PreferenceSchema = {
             type: 'number',
             default: 1
         },
+        'terminal.integrated.shell.windows': {
+            type: 'string',
+            description: 'The path of the shell that the terminal uses on Windows (default: C:\\Windows\\System32\\cmd.exe).',
+            default: 'C:\\Windows\\System32\\cmd.exe'
+        },
+        'terminal.integrated.shell.osx': {
+            type: 'string',
+            description: `The path of the shell that the terminal uses on macOS (default: ${process.env.SHELL || '/bin/bash'}).`,
+            default: process.env.SHELL || '/bin/bash'
+        },
+        'terminal.integrated.shell.linux': {
+            type: 'string',
+            description: `The path of the shell that the terminal uses on Linux (default: ${process.env.SHELL || '/bin/bash'}).`,
+            default: process.env.SHELL || '/bin/bash'
+        },
+        'terminal.integrated.shellArgs.windows': {
+            type: 'array',
+            description: 'The command line arguments to use when on the Windows terminal.',
+        },
+        'terminal.integrated.shellArgs.osx': {
+            type: 'array',
+            description: 'The command line arguments to use when on the macOS terminal.',
+        },
+        'terminal.integrated.shellArgs.linux': {
+            type: 'array',
+            description: 'The command line arguments to use when on the Linux terminal.',
+        },
```

Comment:
We should add a default value for each of these preferences:

```ts
default: []
```

https://github.com/microsoft/vscode/blob/df1f0712f6f8def6e0b52065b6fe66dab66c4e53/src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts#L33-L67

Reply:
Also done